src/main/java/it/gov/pagopa/rtd/ms/rtdmsdecrypter/service/BlobVerifierImpl.java

```diff
@@ -98,4 +116,27 @@ private void logVerificationInformation(String blobName, Long deserializedSize,
     log.info("END Verifying {} records: {} valid , {} malformed", blobName,
         deserializedSize, violations);
   }
+
+  private void gatheringMetadataAndCount(BlobApplicationAware blob, DecryptedRecord decryptedRecord, AtomicLong numberOfDeserializeRecords) {
+    AdeTransactionsAggregate tempAdeAgg = (AdeTransactionsAggregate) decryptedRecord;
+    ReportMetaData reportMetaData = blob.getOriginalBlob().getReportMetaData();
+    reportMetaData.getMerchantList().add(tempAdeAgg.getMerchantId());
+    if (tempAdeAgg.getOperationType().equals("00")) {
+      reportMetaData.setNumPositiveTrx(reportMetaData.getNumPositiveTrx() + tempAdeAgg.getNumTrx());
+      reportMetaData.setTotalAmountPositiveTrx(
+          reportMetaData.getTotalAmountPositiveTrx() + tempAdeAgg.getTotalAmount());
+    } else {
+      reportMetaData
+          .setNumCanceledTrx(reportMetaData.getNumCanceledTrx() + tempAdeAgg.getNumTrx());
```

Comment:
might result in cleaner code something like this? 

```suggestion
          .increaseNumCanceledTrxBy(tempAdeAgg.getNumTrx());
```
of course that method must be implemented in the reportMetadata object

Reply:
Yes, of course, I think it is a great idea, I should apply it also to the other increasing actions. Thx!